nixjs-rt/src/builtins.ts

```diff
@@ -354,12 +358,26 @@ export function getBuiltins() {
         throw builtinBasicTypeMismatchError("import", pathStrict, expected);
       }
 
-      const pathValue = pathStrict.toJs();
+      let pathValue = "";
+      if (pathStrict instanceof NixString) {
+        pathValue = normalizePath(pathStrict.toJs());
+      } else if (pathStrict instanceof Path) {
+        pathValue = pathStrict.toJs();
+      }
+
+      // Check if it's an absolute path. Relative paths are not allowed.
+      // Path data types are always automatically absolute.
+      if (!isAbsolutePath(pathValue)) {
+        throw otherError(
+          err`string ${highlighted(JSON.stringify(pathValue))} doesn't represent an absolute path. Only absolute paths are allowed for imports.`,
+          "builtins-import-non-absolute-path",
```

Comment:
Nitpick: Maybe add a test for this? Feel free to ignore or add in a separate PR.